bin/tx-prover/src/utils.rs

```diff
@@ -1,20 +1,71 @@
 use std::time::Duration;
 
+use opentelemetry::{trace::TracerProvider as _, KeyValue};
+use opentelemetry_sdk::{
+    runtime,
+    trace::{RandomIdGenerator, Sampler, TracerProvider},
+    Resource,
+};
+use opentelemetry_semantic_conventions::{
+    resource::{SERVICE_NAME, SERVICE_VERSION},
+    SCHEMA_URL,
+};
 use pingora::{http::ResponseHeader, Error, ErrorType};
 use pingora_proxy::Session;
 use tonic::transport::Channel;
 use tonic_health::pb::health_client::HealthClient;
+use tracing::Level;
+use tracing_subscriber::{layer::SubscriberExt, Registry};
+
+pub const TRACING_TARGET_NAME: &str = "miden-tx-prover";
 
 const RESOURCE_EXHAUSTED_CODE: u16 = 8;
 
-pub(crate) fn setup_tracing() {
-    // Set a default log level if `RUST_LOG` is not set
-    if std::env::var("RUST_LOG").is_err() {
-        std::env::set_var("RUST_LOG", "info"); // Default to 'info' level
-    }
-    tracing_subscriber::fmt::init();
+// Construct TracerProvider for OpenTelemetryLayer
+pub(crate) fn init_tracer_provider() -> TracerProvider {
+    let exporter = create_span_exporter();
+
+    TracerProvider::builder()
+        .with_sampler(Sampler::ParentBased(Box::new(Sampler::TraceIdRatioBased(1.0))))
+        .with_id_generator(RandomIdGenerator::default())
+        .with_resource(create_resource())
+        .with_batch_exporter(exporter, runtime::Tokio)
+        .build()
+}
+
+// Create a SpanExporter
+fn create_span_exporter() -> opentelemetry_otlp::SpanExporter {
+    opentelemetry_otlp::SpanExporter::builder()
+        .with_tonic()
+        .build()
+        .expect("Failed to build SpanExporter")
+}
+
+// Create a Resource that captures information about the entity for which telemetry is recorded.
+fn create_resource() -> Resource {
+    Resource::from_schema_url(
+        [
+            KeyValue::new(SERVICE_NAME, env!("CARGO_PKG_NAME")),
+            KeyValue::new(SERVICE_VERSION, env!("CARGO_PKG_VERSION")),
+        ],
+        SCHEMA_URL,
+    )
 }
 
+// Setup tracing subscriber
+pub(crate) fn setup_tracing(provider: TracerProvider) -> Result<(), String> {
```

Comment:
Similar to the previous comment - could we add more details about what this function does?

Also, what's the motivation for have this and `init_tracer_provider()` as two separate functions? It seems like one is called right after the other. Should we combine them into one function?